tournaments/index.js

```diff
@@ -672,6 +673,16 @@ Tournament = (function () {
 		this.runAutoDisqualify();
 		this.update();
 	};
+	Tournament.prototype.onBattleJoin = function (room, user) {
+		if (this.scouting || this.isEnded || user.latestIp === room.p1.latestIp || user.latestIp === room.p2.latestIp) return;
+		var roomid = (room && room.id ? room.id : room);
+		var users = this.generator.getUsers(true);
+		for (var i = 0; i < users.length; i++) {
+			if (users[i].latestIp === user.latestIp) {
```

Comment:
Like, it's not like it'd be that much harder to get around if we switched to an O(n^3) algorithm rather than this one.